### PR TITLE
Adding MPI design document

### PR DESCRIPTION
**Pull request by ericbuckley**
## Description
Adding a new documentation file `docs/mpi-design.md` to provide a detailed overview of the Master Patient Index (MPI) system.

## Additional Notes
Also moved the existing `docs/*` into `docs/policies/*` as they aren't very useful for the project itself, just information on CDC's open source policies.

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

**Review by alhayward — DISMISSED** [review]
This is awesome! The language is really accessible in explaining key parts of the MPI and how they relate to the RL algorithm. Left a few questions/suggestions for clarity. Thanks for putting this together!

File: docs/mpi-design.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Overview of MPI Person-Person Linking Models
```
[the remark this answers is not in the document]
Comment (ericbuckley) [reply docs/mpi-design.md:1]: yeah that doesn't read well.  What do you think about?
```suggestion
# Overview of MPI Patient-Patient Linking Models
```

[the remark this answers is not in the document]
Comment (alhayward) [reply docs/mpi-design.md:1]: Tbh, I would just say `# Overview of MPI Models`. What do you think?

[the remark this answers is not in the document]
Comment (alhayward) [reply docs/mpi-design.md:1]: Also thanks for showing me how to suggest changes! 😅

**Review by alhayward — APPROVED** [review]

**Pull request merged**